core/services/chainlink/application.go

```diff
@@ -296,10 +298,31 @@ func NewApplication(opts ApplicationOpts) (Application, error) {
 	promReporter := promreporter.NewPromReporter(db.DB, globalLogger)
 	srvcs = append(srvcs, promReporter)
 
+	// Initialize Sessions ORM based on environment configured authenticator
+	// localDB auth or remote LDAP auth
+	authMethod := cfg.WebServer().AuthenticationMethod()
+	var sessionsORM sessions.UserManager
+	var sessionReaper utils.SleeperTask
+
+	switch authMethod {
+	case string(sessions.LDAPAuth):
+		var err error
+		sessionsORM, err = ldapauth.NewLDAPAuthenticator(
+			db, cfg.Database(), cfg.WebServer().LDAP(), cfg.Insecure().DevWebServer(), globalLogger, auditLogger,
+		)
+		if err != nil {
+			return nil, errors.Wrap(err, "NewApplication: failed to initialize LDAP Authentication module")
+		}
+		sessionReaper = ldapauth.NewLDAPServerStateSync(db, cfg.Database(), cfg.WebServer().LDAP(), globalLogger)
+	default:
```

Comment:
Could we be more explicit here as to what values are allowable and default to the legacy behaviour. At the moment this is baking any value other than `sessions.LDAPAuth` as using localauth. I think it would be better if we added a default to the AuthenticationMethod TOML field and used localauth if we successfully test against that default.

Anything other than LDAPAuth or this default value should result in an error of some kind.

Reply:
I like this idea, especially if future auth systems are added (like OAuth, or other SSO solutions). I think it could also form the basis to possibly organize fields for localauth.

My concern would be scope creep here, so if it's difficult (or requires rearchitecting local auth with new structures for a new variadic type) perhaps we could file this for follow up?